### ISSUE TITLE
Default value not applied

### ISSUE THREAD

**Issue by matthewnessworthy** [thefrontside/emberx-select#81]
Using the simple example with property binding, a normal select would have the first value selected without user interaction. With emberx-select, unless you manually select an option the value `model.status` will be empty/undefined/etc by default.
Can/should this be fixed, or is there a workaround?

```
{{#x-select value=model.status }}
  {{#x-option value=1}}Active{{/x-option}}
  {{#x-option value=2}}Inactive{{/x-option}}
{{/x-select}}
```

**Comment by cowboyd** [comment]
With `emberx-select`, the only way that Active would be checked would be if `model.status` had a value of `1`. Are you suggesting that `{{x-select}}` should initialize `model.status` to the first value if it is undefined?

**Comment by alexphys** [comment]
I think that {{x-select}} should set model.status to the first value not only when it is rendered but also in cases the content of select options changes and the selected value is no longer included in the options.

Imagine a case where you have 2 selects where the second select's options are dependent on the selection of the first select;
```
###First selection 

{{#x-select value=model.parent }}
  {{#each option in array}}
    {{#x-option value=option}}option.name{{/x-option}}
  {{/each}}
{{/x-select}}

###2nd selection with options dependent on the 1st selection

{{#x-select value=model.child }}
  {{#each option in model.parent.children}}
    {{#x-option value=option}}option.name{{/x-option}}
  {{/each}}
{{/x-select}}
```

**Comment by matthewnessworthy** [comment]
I agree with @alexphys , and in general, since this is a select helper/native replacement, it should function in the same way, which means preselecting the initial value as well as selecting the first value in the list should the selected value be removed at a later time

**Comment by sa4breeze** [comment]
The default selected attribute in x-option also has an issue.

Snippet 1:
```
{{#x-select class="form-control" value=SomePerson}}
    {{#x-option value=0 selected}}Alice{{/x-option}}
    {{#x-option value=1}}Bob{{/x-option}}
{{/x-select}}
```

should work similarly to the one shown below:

Snippet 2:
```
<select id=SomePerson>
    <option value=0 selected>Alice<option>
    <option value=1>Bob<option>
</select>
```

In snippet 1, using x-select to pass the value for SomePerson results in undefined. That shouldn't be the case.

It should pass the value as 0 which is set to selected as shown in Snippet 2.

**Comment by mxmz-** [comment]
When this will be fixed? I think this is really a major issue/bug affecting basic functionality of the addon

**Comment by lydiaguarino** [comment]
Hello there
I am currently working on a solution for these issues.

I understand there to be three requests:
1) Fix the broken selected=true behavior for specifying a default selection
2) Provide a way to default to the first option in the list on render
3) Provide a way to default to the first option in the list when the selection is no longer valid because the options in the list have changed

For the time being, we can offer the following solution:
```
{{#x-select value=model.status }}
  {{#x-option value=1}}Active{{/x-option}}
  {{#x-option value=2 selected=true}}Inactive{{/x-option}}
{{/x-select}}
```
Will result in model.status defaulting to value=2 Inactive if model.status is not already set.

```
{{#x-select value=model.status }}
  {{#each choices as | choice index |}}
    {{#if index}}
      {{#x-option value=choice.value}}{{choice.name}}{{/x-option}}
    {{else}}
      {{#x-option value=choice.value selected=true}}{{choice.name}}{{/x-option}}
    {{/if}}
  {{/each}}
{{/x-select}}
```
Will result in model.status defaulting to the first option in the list if model.status is not already set. This will also update correctly if the content of choices changes.

Would this solution satisfy your use cases until we can implement a more elegant solution?

**Comment by lydiaguarino** [comment]
This was addressed in #90 to be released in 2.1
